File: 1 курс/Задачи на работу с текстовыми файлами №1/main.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main() {
    //1
    ifstream file("numbers.txt");
    double num, sum = 0;
    int count = 0;
    while (file >> num) {
        sum += num;
        count++;
    }
    file.close();
    double average = sum / count;
    cout << "Среднее арифметическое: " << average << endl;

    //2
    ifstream file("numbers.txt");
    double num, min, max;
    file >> num;
    min = max = num;
    while (file >> num) {
        if (num < min) min = num;
        if (num > max) max = num;
    }
    file.close();
    cout << "Min: " << min << "\nMax: " << max << endl;

    //3
    ifstream inputFile("numbers.txt");
    ofstream outputFile("even_numbers.txt");
    int number;
    while (inputFile >> number) {
        if (number % 2 == 0) {
            outputFile << number << ' ';
        }
    }

    //4
    ifstream file("text.txt"); 
    int lineCount = 0;
    string line;
    while (getline(file, line)) { 
        lineCount++;  
    }
    cout << "Количество строк: " << lineCount << endl;

    //5
    ifstream file("My_text.txt");  
    int wordCount = 0;
    string word;
    while (file >> word) {
        wordCount++;
    }
    cout << "Количество слов: " << wordCount << endl;

    //6
    string filename = "text.txt";
    string old_word, new_word;

    cout << "Введите слово для замены: ";
    getline(cin, old_word);
    cout << "Введите новое слово: ";
    getline(cin, new_word);
    ifstream inFile(filename);
    string content((istreambuf_iterator<char>(inFile)), {});
    inFile.close();
    size_t pos = 0;
    while ((pos = content.find(old_word, pos)) != string::npos) {
        content.replace(pos, old_word.length(), new_word);
        pos += new_word.length();
    }
    ofstream outFile(filename);
    outFile << content;
    outFile.close();

    return 0;
}

File: 1 курс/Задачи на работу с текстовыми файлами №1/main.py
#1
filename = 'numbers.txt'
sum = 0
count = 0
with open(filename, 'r') as file:
    for line in file:
        parts = line.split()
        for part in parts:
            sum += float(part)
            count += 1
average = sum / count if count > 0 else 0
print(f"Среднее арифметическое: {average}")

#2
filename = 'numbers.txt'
numbers = []
with open(filename, 'r') as file:
    for line in file:
        parts = line.split()
        for part in parts:
            if part.isdigit():
                numbers.append(float(part))
if numbers:
    min_num = min(numbers)
    max_num = max(numbers)
    print(f"Наименьшее число: {min_num}\nНаибольшее число: {max_num}")
else:
    print("В файле нет чисел.")

#3
input_file = 'numbers.txt'
output_file = 'even_numbers.txt'
with open(input_file, 'r') as infile, open(output_file, 'w') as outfile:
    for line in infile:
        parts = line.split()
        for part in parts:
            if part.isdigit() and int(part) % 2 == 0:
                outfile.write(part + '\n')

#4
filename = 'text.txt'
line_count = 0
with open(filename, 'r') as file:
    for _ in file:
        line_count += 1
print(f"Количество строк: {line_count}")

#5
filename = 'My_text.txt'
word_count = 0
with open(filename, 'r') as file:
    for line in file:
        words = line.split()
        word_count += len(words)
print(f"Количество слов: {word_count}")

#6
filename = 'text.txt'
old_word = input("Введите слово для замены: ")
new_word = input("Введите новое слово: ")
with open(filename, 'r') as file:
    content = file.read()
updated_content = content.replace(old_word, new_word)
with open(filename, 'w') as file:
    file.write(updated_content)
